nni/algorithms/compression/v2/pytorch/base/scheduler.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import gc
+import logging
+import os
+from pathlib import Path
+from typing import List, Dict, Tuple, Literal, Optional
+
+import json_tricks
+import torch
+from torch.nn import Module
+from torch.tensor import Tensor
+
+_logger = logging.getLogger(__name__)
+
+
+class Task:
+    _reference_counter = {}
+
+    def __init__(self, task_id: int, model_path: str, masks_path: str, config_list_path: str) -> None:
+        """
+        Parameters
+        ----------
+        task_id
+            The unique id of task.
+        model_path
+            The path of the unwrapped pytorch model that will be pruned in this task.
+        masks_path
+            The path of the masks that applied on the model before pruning.
+        config_list_path
+            The path of the config list that used in this task.
+        """
+        self.task_id = task_id
+        self.model_path = model_path
+        self.masks_path = masks_path
+        self.config_list_path = config_list_path
+
+        self.status: Literal['Pending', 'Running', 'Finished'] = 'Pending'
+        self.score: Optional[float] = None
+
+        self.state = {}
+
+        for ref in self.referenced_paths():
+            self._reference_counter.setdefault(ref, 0)
```

Comment:
Is `_reference_counter`  defined in the whole class the same object with `self._reference_counter ` here?

Reply:
We directly set the reference count to zero here what if `ref` is also used by other threads?

Reply:
yes, all `self._reference_counter` in different instances refer to the same dict `_reference_counter` in class.

it is a good question, maybe we need to use some kind of lock if support multi-threads. for now, we run pruning in a single thread, and I will add comments in this part to remind us, need to refactor this part if we want to support multi-threads.